src/Integration.UnitTests/LocalServices/RuleSetInspectorTests.cs

```diff
@@ -0,0 +1,494 @@
+            this.shell.RegisterPropertyGetter((int)__VSSPROPID2.VSSPROPID_InstallRootDir, () => this.SonarQubeRuleSetFolder);
+            this.serviceProvider.RegisterService(typeof(SVsShell), this.shell);
+
+            this.testSubject = new RuleSetInspector(this.serviceProvider);
+
+            Directory.CreateDirectory(this.VsRuleSetsDirectory);
+
+            this.temporaryFiles = new TempFileCollection();
+        }
+
+        [TestCleanup]
+        public void TestCleanup()
+        {
+            // Catch release-build issues that would otherwise be ignored because Debug.Assert will not be called
+            this.outputPane.AssertOutputStrings(0);
+            ((IDisposable)this.temporaryFiles).Dispose();
+        }
+        #endregion
+
+        #region Properties
+        /// <summary>
+        /// Simulates the solution level SonarQube folder in which we store the fetched rulesets
+        /// </summary>
+        private string SonarQubeRuleSetFolder
+        {
+            get
+            {
+                return this.TestContext.TestRunDirectory;
+            }
+        }
+
+        /// <summary>
+        /// Simulates the usages of the vs ruleset folder for the default rulesets that ship with VS
+        /// </summary>
+        private string VsRuleSetsDirectory
+        {
+            get
+            {
+                return Path.Combine(this.TestContext.TestRunDirectory, RuleSetInspector.DefaultVSRuleSetsFolder);
+            }
+        }
+        #endregion
+
+        #region Tests
+        [TestMethod]
+        public void RuleSetInspector_FindConflictingRules_ProjectLevelOverridesOfTheSolutionRuleset()
+        {
+            // Setup
+            string solutionRuleSet = this.CreateCommonRuleSet().FilePath;
+            this.temporaryFiles.AddFile(solutionRuleSet, false);
+
+            // Check all supported RuleAction values
+            RuleAction[] unsupportedActions = new[] { RuleAction.Default };
+            foreach (RuleAction ruleAction in Enum.GetValues(typeof(RuleAction)).OfType<RuleAction>().Except(unsupportedActions))
+            {
+                foreach (IncludeType includeType in Enum.GetValues(typeof(IncludeType)).OfType<IncludeType>())
+                {
+                    this.TestContext.WriteLine("Running test case, Project Rules are {0}, SolutionInclude is {1}", ruleAction, includeType);
+
+                    RuleSet projectRuleSet = CreateProjectRuleSetWithInclude(DefaultNumberOfRules, solutionRuleSet, includeType, ruleAction);
+                    this.temporaryFiles.AddFile(projectRuleSet.FilePath, false);
+
+                    // Act
+                    RuleConflictInfo conflicts = this.testSubject.FindConflictingRules(solutionRuleSet, projectRuleSet.FilePath);
+
+                    // Verify
+                    if (ruleAction == RuleAction.None)
+                    {
+                        AssertMissingRulesByFullIds(conflicts, projectRuleSet.Rules);
```

Comment:
Logically, shouldn't the test be checking that all of the rules the _solutionRuleSet_ have been removed, rather than passing in the _projectRuleSet_? It makes the test harder to understand.

Presumably the test is passing because the _projectRuleSet_ and _solutionRuleset happen to be the same. It's not obvious why this is the case from looking at this method; the project ruleset is constructed by explicitly passing the number of rules, whereas the solution ruleset is using a default parameter.